src/components/forms/DateInput/DateInput.tsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react'
+import classnames from 'classnames'
+
+import { TextInput } from '../TextInput/TextInput'
+import { Label } from '../Label/Label'
+import { Fieldset } from '../Fieldset/Fieldset'
+
+interface DateInputElementProps {
+  id: string
+  name: string
+  label: string
+  maxLength: number
+  minLength?: number
+}
+
+interface DateInputProps {
+  id: string
+  name: string
+  legend: React.ReactNode
+  month?: boolean
+  day?: boolean
+  year?: boolean
+  hint?: string
+  className?: string
+}
+
+const DateInputElement = (
+  props: DateInputElementProps & React.HTMLAttributes<HTMLElement>
+): React.ReactElement => {
+  const { id, name, label, maxLength, minLength } = props
+
+  return (
+    <div className="usa-form-group usa-form-group--month">
```

Comment:
This class needs to change based on which input it is (`usa-form-group--day`, etc.). Also there is a `forms/FormGroup` component I think you could use here.